feature-utils/poly-look/src/react-components/overlays/sideSlider.jsx

```diff
@@ -0,0 +1,422 @@
+import React, { useEffect, useState, useRef } from "react";
+
+import "./sideSlider.css";
+
+let stPoz;
+let observer;
+let alphaInc = {
+  x: null,
+  y: null,
+};
+let width;
+
+const steps = {
+  in: "in",
+  interactive: "interactive",
+  out: "out",
+  outUp: "outUp",
+  outDone: "outDone",
+};
+const axes = {
+  x: "x",
+  y: "y",
+};
+const directions = {
+  right: "right",
+  left: "left",
+  up: "up",
+  down: "down",
+};
+
+/**
+ *
+ * Generic container for HTRT side sheets or any other component that
+ * slides from the right of the screen and can be closed either through
+ * click, right or up sweep.
+ * Note that during the in/out animations, interactions are disabled.
+ * @param {Object} props
+ * @param {boolean} [props.open] - Open or close the slide;
+ * defaults to false.
+ * @param {callback} [props.onClose] - Called after the exit animations end.
+ * @param {string} [props.leftDistance] - Contents distance from the left
+ * of the screen. Can be any value that is compatible with CSS calc rule;
+ * defaults to 124px.
+ * @param {Array[number]} [props.backdropColor] - The backdrop color as a rgba array;
+ * defaults to [0, 0, 0, 0.3].
+ * @param {string} [props.screenMaxWidth] - The max width of the screen;
+ * defaults to the value of --max-width.
+ * @param {string} [props.animationDuration] - The duration of the animations.
+ * Can be any value that is compatible with CSS transition-duration rule;
+ * defaults to 0.6s.
+ * @param {JSX.Element} [props.Component] - The component that will be animated.
+ * If this component can trigger the SideSlider to close then it must have
+ * an onClose callback as one of it's props.
+ * If it has clickable elements, then Event.stopPropagation()
+ * must be called in the event's handler function, otherwise the Slider will close.
+ * If the component has vertical scroll then it must accept custom styles.
+ * @param {string} [props.lastChildSelector] - CSS selector for the last child of
+ * the Component prop. Use it if you think the Component might have
+ * a vertical scroll, unnecessary otherwise.
+ * @returns {JSX.Element}
+ */
+
+const SideSlider = ({
+  open = false,
+  onClose,
+  leftDistance = "124px",
```

Comment:
You mean, something like having them in `constants.js`? I'm not sure it's worth it. My plan is to update the defaults if something needs changing after we use it in the features (I'm thinking `leftDistance` and `backdropColor` are very probable candidates).

Reply:
Well, that's one option, but if they're local you can simply declare them at the beginning of the file. The point is to avoid literals as much as possible, and substitute them with constants with a meaningful name.

Reply:
So, currently this is the only change that I haven't addressed. This is because I still don't understand what we would gain by doing this. The requirement of them being centrally changed is fulfilled but we don't have extra variables.